docker/gaffer-pyspark-notebook/examples/gaffer-rest-api-road-traffic-example.ipynb

```diff
@@ -0,0 +1,361 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Configuration\n",
+    "If the following variable is left blank then we will attempt to discover the correct URL from the notebook's environment"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "GAFFER_REST_API_URL = ''"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Imports"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import io\n",
+    "import json\n",
+    "import pandas as pd\n",
+    "import os\n",
+    "\n",
+    "!pip install plotly\n",
+    "import plotly.graph_objects as go\n",
+    "\n",
+    "import gafferpy\n",
+    "from gafferpy import gaffer\n",
+    "from gafferpy import gaffer_connector\n",
+    "\n",
+    "pd.set_option('display.max_rows', None)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Set up connection to Gaffer REST API"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Try to auto-detect REST API URL from environment\n",
+    "if not GAFFER_REST_API_URL:\n",
+    "    try:\n",
+    "        GAFFER_REST_API_URL = os.environ['GAFFER_REST_API_URL']\n",
+    "    except KeyError:\n",
+    "        pass\n",
+    "\n",
+    "if not GAFFER_REST_API_URL:\n",
+    "    raise Exception('Please set GAFFER_REST_API_URL')\n",
+    "print('Using Gaffer REST API URL: {}'.format(GAFFER_REST_API_URL))\n",
+    "\n",
+    "gc = gaffer_connector.GafferConnector(GAFFER_REST_API_URL)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Get Graph Schema"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "schema_response = gc.execute_get(gaffer.GetSchema())\n",
```

Comment:
Could perhaps also get the schema from `/etc/gaffer/schema` just to demonstrate?

Reply:
Yeah could do. The reason why I did it this way is that it also serves as a test to check that the REST API is up and the notebook is able to access it, and it doesn't require the backing store to be functional for a successful response. So if this operation works, but the next ones fail, then we know that a problem lies between the REST API and the backing store.

Reply:
Perhaps another block could be added to print the schema from the locally mounted file also? Not only would this test it gets mounted but be a good way to show how it works? Maybe even assert they are the same?

Reply:
👍 done

Reply:
Awesome, thanks!